.docker/Dockerfile-hsm

```diff
@@ -53,30 +53,36 @@ ENV HSM_LIBRARY=/usr/lib/softhsm/libsofthsm2.so
 ENV HSM_TOKEN_LABEL=hydra
 ENV HSM_PIN=1234
 
+# NOTE: This is broken already. Even though this image provides a shell, you'd need to configure it with
+# `SHELL ["/busybox/sh", "-c"]`, however `apt-get` does not exist either in a distroless image.
+# This was original an Alpine image, the refactoring was not verified properly in this commit:
+# https://github.com/ory/hydra/commit/c1e1a569621d88365dceee7372ca49ecd119f939#diff-ae54bef08e3587b28ad8e93eb253a9a5cd9ea6f4251977e35b88dc6b42329e25L31
```

Comment:
The HSM image is really just to run some e2e hsm tests. It's not being distributed and should not be used.

Reply:
Yeah, e2e tests are now failing. Probably just revert the changes here:  https://github.com/ory/hydra/actions/runs/12523819546/job/34933781514?pr=3914